crates/optimism/payload/src/payload.rs

```diff
@@ -287,10 +289,17 @@ where
 
         let parent_beacon_block_root = block.parent_beacon_block_root.unwrap_or_default();
 
+        let payload_v3 = ExecutionPayloadV3::from_block_unchecked(
+            block.hash(),
+            &Arc::unwrap_or_clone(block).into_block(),
+        );
+        // todo: return error if v4 block doesn't contain l2 withdrawals root
+        let l2_withdrawals_root = block.withdrawals_root.unwrap_or_default();
```

Comment:
I think this should be fine because this enforced based on the endpoint via the enginetypes I believe